File: backend/schemas/visit.py
from datetime import datetime
from typing import Optional,List
from pydantic import BaseModel
from schemas.medicine import MedicineResponse
from schemas.symptom import SymptomResponse

class VisitBase(BaseModel):
    diagnosis: str
    comments : str
    reason : str
    examination : str
    control : str | None
    
    class Config:
        orm_mode = True

class VisitFileBase(BaseModel):
    id: int
    storage_key: str
    original_name: str
    created_at: datetime

    class Config:
        orm_mode = True

class VisitResponseForVisitor(BaseModel):
    id:int
    diagnosis:str
    created_at:datetime

class VisitCreate(VisitBase):
    reason : str
    examination : str
    control : str | None

class VisitResponse(VisitBase):
    id:int
    created_at: datetime
    files: list[VisitFileBase] = []
    medicines: Optional[List[MedicineResponse]]
    symptoms : Optional[List[SymptomResponse]]
    

File: frontend/src/types/visit.ts
export type IdRef = { id: number };

export type VisitInput = {
  diagnosis: string;
  comments: string;
  reason: string;
  examination: string;
  control: string;
  medicines: IdRef[];
  symptoms: IdRef[];
};

export type CreateVisitPayload = {
  new_visit: {
    diagnosis: string;
    comments: string;
    reason: string;
    examination: string;
    control: string | null;
  };
  symptoms: IdRef[];
  medicines: IdRef[];
};

export type VisitFileRef = { id: number; name?: string | null };

export type VisitDetail = {
  id: number;
  diagnosis: string;
  comments: string;
  reason: string;
  examination: string;
  control: string | null;
  created_at: string;
  files: VisitFileRef[];
  medicines: { id: number; name: string }[];
  symptoms: { id: number; name: string }[];
};
